public/app/features/explore/TraceView/components/TracePageHeader/SpanFilters/SpanFilters.tsx

```diff
@@ -323,9 +152,10 @@ export const SpanFilters = memo((props: SpanFilterProps) => {
                 isClearable
                 onChange={(v) => setSpanFiltersSearch({ ...search, serviceName: v?.value || '' })}
                 onOpenMenu={getServiceNames}
-                options={serviceNames}
+                options={serviceNames || (search.serviceName ? [search.serviceName].map(toOption) : [])}
```

Comment:
This creates a scenario where if you have selected a service name in the options, then it is the only service name you can select in the span filters.